ios/ReactNativeBase/AppDelegate.m

```diff
@@ -39,6 +40,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
   RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge
                                                    moduleName:@"ReactNativeBase"
                                             initialProperties:nil];
+  [RNSplashScreen show];
```

Comment:
Maybe with
```suggestion
  [RNSplashScreen show];
  return YES;
```

the timeout inside the useEffect on the App.js would not be necessary.  I didn't test it just saw it on another guide. That's the only difference between your code to avoid the white flash on the splash from RN



Reference image
![image](https://user-images.githubusercontent.com/57040392/126574636-74df894a-c865-4d38-9a2f-a2f5d538dfd5.png)

Reply:
Ohhh I did the timeout solution in my project xD I'm gonna try this! thanks!

Reply:
Well I just saw and I have the return YES already 😢

Reply:
You have the [RNSplashScreen show]; at the end of the launch options function ? next to the return YES; ? that's what I meant

Reply:
yes 

![image](https://user-images.githubusercontent.com/8755889/126656971-9c673aaf-3ce9-419b-b420-8d4e99fce842.png)